Rename default section name and include section on login card

davidwufer opened the pull request:
@Erin007 @tanyaparker FYI

Spec: https://docs.google.com/document/d/1NSHc0HF6xIaFQyaSMuz2qMpvlSM09HTu3O3uW_UBkWk/edit?ts=5b60ef3d#heading=h.wfpv5zo4wzbq

davidwufer commented on lib/cdo/shared_constants.rb:
```diff
@@ -38,6 +38,13 @@ module SharedConstants
   ).freeze
 
+  # The set of default values for a section
+  SECTION_DEFAULTS = OpenStruct.new(
```
I wasn't entirely sure how to use this since it's in `lib/cdo/shared-constants.rb` and I'd like to avoid hardcoding the string in the places above. Any suggestions?

Erin007 replied:
1.) You'll need to add `SECTION_DEFAULTS` to the list in  the script that generates shared constants, https://github.com/code-dot-org/code-dot-org/blob/aa6a61a400862c70b1de0fb03e38d644e51621b4/apps/script/generateSharedConstants.rb#L73. 
2.) Then, you should have access to your constant with this pattern: https://github.com/code-dot-org/code-dot-org/blob/06ae58fdd52a97d779aff24db753dabc8d4afd97/pegasus/helpers/section_api_helpers.rb#L256

Erin007 commented:
First, you'll need to remove `locale/dsls.en.yml` and `db/schema_cache.dump` from this PR.  They are the "hitchhikers" I referred to in our meeting. 🙂 You can do that with `git checkout <filename>` to restore them to their original version so there isn't a diff with staging. (More info:[ here](https://git-scm.com/docs/git-checkout)) If you get tangled don't hesitate to reach out or join our #git-fu Slack channel with any questions. This has happened to me too,  and you can prevent in the future by  running `git status` before committing to see which files are going to be included, then only adding the exact ones you've changed.

Erin007 commented:
@davidwufer I'm going to loop in our internationalization gurus since we ultimately want "Untitled Section" to be a translatable string. 

@Hamms and @nkiruka, this is David, a contributor who's working on changing the default section name from "New Section" to "Untitled Section" and adding section names to printable login cards. 

Where should we put the "Untitled Section" string so that it can both be translated and accessible where he needs it for this change? Many thanks!

Hamms commented:
Hello David, thanks for your contribution!

To add a new i18n string in dashboard, you'll want to add it to the appropriate place in `dashboard/config/locales/en.yml`; I'd recommend adding something like `default_name: "Untitled Section"` somewhere in the `section` section: https://github.com/code-dot-org/code-dot-org/blob/cd646c98667659b14887165801b4d52a2ea71664/dashboard/config/locales/en.yml#L240-L253

You can then reference it as `I18n.t('sections.default_name')`, from both dashboard and pegasus code.

davidwufer commented:
@Hamms @Erin007 Thank you both. I decided to go with the approach of adding an entry for `'Untitled Section'` to be translatable, and to change existing labels to call the i18n library with a default hardcoded value. That way when the string gets translated, we can go through the different places the internationalized string is used and delete the default value if needed. This seems more lightweight than using `shared_constants.rb`. Let me know what you think!

I removed the hitchhikers. Sorry, I thought I had removed them before submitting the first PR 😞

islemaster commented:
Thanks for contributing!

sureshc commented:
@davidwufer Thanks very much for contributing these improvements! :)

davidsbailey commented:
Thanks for implementing this David, this is a big help!
